tsconfig.json

```diff
@@ -17,6 +17,7 @@
     "noImplicitReturns": true,
     "noUnusedLocals": true,
     "noUnusedParameters": true,
+    "esModuleInterop": true
```

Comment:
What's this do? I'm assuming something related to the import changes.

Reply:
I mean I read https://www.typescriptlang.org/tsconfig#esModuleInterop but need a simpler explanation 💁

Reply:
Basically, it lets you use ES6 module `import` syntax including destructuring. Typescript/ESBuild will auto-convert this to `require` when building to commonjs (hence the interop).

It makes defining the imports much easier, allows use of the destructured syntax for imports, and for esbuild, it allows tree-shaking to occur to leave out bits of code you don't use in a library.

Reply:
It helps to start with a primer on the es6 module syntax first: https://www.digitalocean.com/community/tutorials/js-modules-es6

And then just know this setting polyfills that syntax back to commonjs module format transparently.